wordpress-mobile/WordPress-iOS
# Plan Details: replace No Results view

Fixes #9596 

This replaces the No Results view for Plan Details.

The No Results view is displayed in these cases:
- Loading plan detail
- Loading error
- No connection (this is a new addition)

To test:

This can be difficult to test because the plan details are loaded so fast. So I've included a hack for each scenario in case it's needed.

---

**Loading plan detail:**
Hack: 
- In `PlanDetailViewModel:tableViewModel` always return `ImmuTable.Empty`
- In `PlanDetailViewModel:noResultsViewModel` return the "Loading Plan..." model for the `ready` case.

Test:
- Go to site details > Plans > select a Plan.
- Verify the initial view is:
<img width="250" alt="loading_plan" src="https://user-images.githubusercontent.com/1816888/41492413-3ebbc682-70bc-11e8-9b55-b7edc7cd9ed3.png">

---

**Loading error:**
Hack: 
- In `PlanDetailViewModel:tableViewModel` always return `ImmuTable.Empty`
- In `PlanDetailViewModel:noResultsViewModel` return the "Oops" model for the `ready` case.

Test:
- Go to site details > Plans > select a Plan.
- Verify the view is:
<img width="250" alt="loading_error" src="https://user-images.githubusercontent.com/1816888/41492459-7f118d7a-70bc-11e8-8272-cda7f2c8c5b0.png">

---

**No connection:**
Hack: 
- In `PlanDetailViewModel:tableViewModel` always return `ImmuTable.Empty`
- In `PlanDetailViewModel:noResultsViewModel` return the "No connection" model for the `ready` case.

Test:
- Go to site details > Plans.
- Disable internet connection (unless you've implemented the hack).
- Select a Plan.
- Verify the view is:
<img width="250" alt="no_connection" src="https://user-images.githubusercontent.com/1816888/41492555-0eab62da-70bd-11e8-9705-632258475522.png">
